packages/headless/src/chat.ts

```diff
@@ -132,6 +146,51 @@ type Action =
 
 export type OnMessage = (action: Action) => void;
 
+export function useFindUncontactedTutors(): {
+  query: UseInfiniteQueryResult<
+    InfiniteData<Paginated<ITutor.UncontactedTutorInfo>, unknown>,
+    Error
+  >;
+  list: ITutor.UncontactedTutorInfo[] | null;
+  more: () => void;
+} {
+  const atlas = useAtlas();
+
+  const findUncontactedTutors = useCallback(
+    ({ pageParam }: { pageParam: number }) => {
+      return atlas.user.findUncontactedTutors({ page: pageParam });
+    },
+    [atlas.user]
+  );
+
+  return useInfinitePaginationQuery(findUncontactedTutors, [
+    QueryKey.FindUncontactedTutors,
+  ]);
+}
+
+export function useCreateRoom({
```

Comment:
The chat room should be created when sending the first message. That's why `atlas.chat.createRoom` accepts optional message text.